taxonomy/api.py

```diff
@@ -1,252 +1,54 @@
 """APIs for lore taxonomy application"""
 from __future__ import unicode_literals
 
-from django.db import transaction
-from django.shortcuts import get_object_or_404
-
 from taxonomy.models import (
     Vocabulary,
     Term,
 )
-from learningresources.models import (
-    LearningResource,
-    LearningResourceType,
+from learningresources.api import (
+    get_repo,
+    NotFound,
 )
 
 
-def get_vocabulary(vocabulary_id):
+def get_vocabulary(repo_slug, user_id, vocab_slug):
     """
-    Lookup vocabulary given its id.
+    Lookup vocabulary given its slug, using repo_slug to validate ownership.
 
     Args:
-        vocabulary_id (int): Vocabulary id
+        repo_id (int): Repository id
+        user_id (int): User id
+        vocab_slug (unicode): Vocabulary slug
 
     Returns:
         Vocabulary (Vocabulary): The vocabulary from the database
     """
-    return get_object_or_404(Vocabulary, id=vocabulary_id)
+    repo = get_repo(repo_slug, user_id)
+    try:
+        return repo.vocabulary_set.get(slug=vocab_slug)
+    except Vocabulary.DoesNotExist:
+        raise NotFound()
 
 
-# pylint: disable=too-many-arguments
-@transaction.atomic
-def create_vocabulary(
```

Comment:
Yeah, my rational is mostly YAGNI. The `Vocabulary.objects.create` lines are all in the tests. These API functions were created for the REST API to make use of so they were removed when they were left unused. If we end up needing this functionality later on we can add it back with support for authorization.